content/blogposts/2022/release-4.2.md

```diff
@@ -0,0 +1,74 @@
+
+In 4.2, Sourcegraph now ships with an easy-to-consume audit log. The log contains security events, Gitserver access events, and GraphQL requests. You can [read more about the audit log in our docs](https://docs.sourcegraph.com/admin/audit_log). The audit log is available for Sourcegraph self-hosted instances, with availability coming soon for Sourcegraph Cloud.
+
+<br />
+<Badge link="/batch-changes" text="Batch Changes" color="blue" size="small" />
+
+#### Secrets in server-side Batch Changes (Beta)
+
+It's common to use secrets in batch changes steps. Developers can use secrets in batch changes to authenticate to a private registry to install packages, create tickets from within a batch change, or make authenticated API calls to other services. In local runs, secrets can be either hardcoded in the batch spec or loaded from environment variables using [`step.env`](https://docs.sourcegraph.com/batch_changes/references/batch_spec_yaml_reference#steps-env), but until now, there was no robust and secure way to manage secrets for server-side runs.
+
+To solve this, we're releasing [executor secrets](https://docs.sourcegraph.com/admin/executor_secrets). You can now define secrets to be passed to server-side runs, and those secrets can be referenced as `env` variables in the batch change spec.
+
+Sourcegraph supports two types of secrets: namespaced secrets that can only be accessed by their owner, and global secets that site admins can set and make available to all users on an instance.
+
+This feature is in Beta, and feedback is very welcome. Tweet at us, or drop a comment in this [issue](https://github.com/sourcegraph/sourcegraph/issues/44597)!
+
+<br />
+<Badge link="/code-search" text="Code Search" color="cerise" size="small" />
+
+#### Smart Search helps with search queries to find more results, faster
+
+To help you find information in your codebase faster than ever, we're introducing Smart Search, a new query assistant that activates when a search returns no results. It can be turned on with the new lightning bolt toggle <span style={{display: "inline-flex", verticalAlign: "middle", margin: "2px"}}><img style={{width: "18px", height: "18px"}} src="https://storage.googleapis.com/sourcegraph-assets/about.sourcegraph.com/blog/2022/smart-search-bar-lightning.png"/></span> in the search bar.
+
+Smart Search helps find search results that are likely to be more useful than showing "no results" by trying slight variations of a user's original query. Smart Search automatically tries alternative queries based on a handful of rules (we know how easy it is to get tripped up by query syntax). When a query alternative finds results, those results are shown immediately. [See the documentation](https://docs.sourcegraph.com/code_search/explanations/features#smart-search) to learn more about Smart Search behavior and configuration.
```

Comment:
@iskyOS I've factored out the example and other details, the link here will be live when https://github.com/sourcegraph/sourcegraph/pull/44778 is merged.

Reply:
merged

Reply:
Thank you!